File: attributes-common/src/main/java/io/github/solusmods/eternalcore/attributes/impl/network/EternalCoreAttributeNetwork.java
package io.github.solusmods.eternalcore.attributes.impl.network;

import io.github.solusmods.eternalcore.attributes.impl.network.c2s.RequestGlideStartPacket;
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils;

public class EternalCoreAttributeNetwork {
    public static void init() {
        NetworkUtils.registerC2SPayload(RequestGlideStartPacket.TYPE, RequestGlideStartPacket.STREAM_CODEC, RequestGlideStartPacket::handle);
    }
}

File: attributes-common/src/main/kotlin/io/github/solusmods/eternalcore/attributes/impl/network/EternalCoreAttributeNetwork.kt
package io.github.solusmods.eternalcore.attributes.impl.network

import dev.architectury.networking.NetworkManager
import io.github.solusmods.eternalcore.attributes.impl.network.c2s.RequestGlideStartPacket
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils.registerC2SPayload
import net.minecraft.network.RegistryFriendlyByteBuf
import net.minecraft.network.codec.StreamCodec
import net.minecraft.network.protocol.common.custom.CustomPacketPayload

object EternalCoreAttributeNetwork {
    fun init() {
        registerC2SPayload(
            RequestGlideStartPacket.Companion.TYPE as CustomPacketPayload.Type<RequestGlideStartPacket?>,
            RequestGlideStartPacket.Companion.STREAM_CODEC as StreamCodec<in RegistryFriendlyByteBuf?, RequestGlideStartPacket?>?
        ) { obj: RequestGlideStartPacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
    }
}